source/lac/dynamic_sparsity_pattern.cc

```diff
@@ -353,6 +347,28 @@ DynamicSparsityPattern::max_entries_per_row() const
 
 
 
+void
+DynamicSparsityPattern::add_row_entries(
+  const size_type &                 row,
+  const ArrayView<const size_type> &columns,
+  const bool                        indices_are_sorted)
+{
+  add_entries(row, columns.begin(), columns.end(), indices_are_sorted);
+}
+
+
+
+void
+DynamicSparsityPattern::add_entries(const ArrayView<const size_type> &rows,
+                                    const ArrayView<const size_type> &columns)
+{
+  AssertDimension(rows.size(), columns.size());
+  for (std::size_t i = 0; i < rows.size(); ++i)
```

Comment:
Again, would it make sense to introduce another variable for the end condition to avoid calling `size()` (here and in the other `for`-loops)?

Reply:
Do you think this might make some difference in the generated code? I assumed this (like the others) would always be inlined.

Reply:
I remember that we were occasionally seeing the end condition to be evaluated for every loop iteration and we had a pull request that tried to fix all of them at once.